pkg/conversion/scheme.go

```diff
@@ -64,19 +60,21 @@ func NewScheme() *Scheme {
 		typeToGVK: map[reflect.Type][]unversioned.GroupVersionKind{},
 		converter: NewConverter(),
 		cloner:    NewCloner(),
-		// TODO remove this hard coded list.  As step one, hardcode it here so this pull doesn't become even bigger
-		InternalVersions: map[string]unversioned.GroupVersion{
-			"":                {},
-			"componentconfig": {Group: "componentconfig"},
-			"extensions":      {Group: "extensions"},
-			"metrics":         {Group: "metrics"},
-		},
-		MetaFactory: DefaultMetaFactory,
 	}
 	s.converter.nameFunc = s.nameFunc
 	return s
 }
 
+// AddInternalGroupVersion registers an internal version to the scheme
```

Comment:
hm, shouldn't it be able to tell internal or not based on the GroupVersion's version field? Why the separate method?